VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionResponsesFragment.java

```diff
@@ -81,9 +81,8 @@ public void onViewCreated(View view, Bundle savedInstanceState) {
                 discussionThread.getIdentifier(),
                 discussionThread.getType() == DiscussionThread.ThreadType.QUESTION);
 
-        // Using application context to prevent activity leak since adapter is retained across config changes
         courseDiscussionResponsesAdapter = new CourseDiscussionResponsesAdapter(
-                getActivity().getApplicationContext(), this, discussionThread);
+                getActivity(), this, discussionThread);
```

Comment:
The comment is misleading; we recreate the adapter on config changes (as this code is in `onViewCreated` which will be called on config change)

Anyway, @1zaman suggestions all sound good to me.